Sources/SwiftSDKGenerator/Queries/CMakeBuildQuery.swift

```diff
@@ -0,0 +1,35 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2022-2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import GeneratorEngine
+import struct SystemPackage.FilePath
+
+@Query
+struct CMakeBuildQuery {
+  let sourcesDirectory: FilePath
+  let outputBinarySubpath: [FilePath.Component]
```

Comment:
```suggestion
  let outputBinarySubpath: FilePath
```

Reply:
Not sure how suitable `FilePath` is for storing relative paths, `[FilePath.Component]` seemed to be a better options for this.

Reply:
Yes, I was reading the docs for FilePath just now and I wondered if that was what you were intending to do, to make sure components such as `bin/lld` aren't mixed up with full paths.